flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunnerTest.java

```diff
@@ -37,44 +43,57 @@
  */
 public class TaskManagerRunnerTest extends TestLogger {
 
-	@Test
-	public void testTaskManagerRunnerShutdown() throws Exception {
-		final Configuration configuration = new Configuration();
-		final ResourceID taskManagerResourceId = ResourceID.generate();
-
-		final ServerSocket localhost = new ServerSocket(0);
-
-		configuration.setString(JobManagerOptions.ADDRESS, localhost.getInetAddress().getHostName());
-		configuration.setInteger(JobManagerOptions.PORT, localhost.getLocalPort());
-		configuration.setString(TaskManagerOptions.REGISTRATION_TIMEOUT, "10 ms");
-		final CompletableFuture<Void> jvmTerminationFuture = new CompletableFuture<>();
-		final TestingTaskManagerRunner taskManagerRunner = new TestingTaskManagerRunner(configuration, taskManagerResourceId, jvmTerminationFuture);
+	@Rule
+	public final Timeout timeout = Timeout.seconds(30);
 
-		taskManagerRunner.start();
+	private Configuration configuration;
```

Comment:
having this as a field seems unnecessary; I'd rather have a method that creates and returns this configuration.

Reply:
Ok, changed.